### PR TITLE
add subscriber to process families for dmf

### PR DESCRIPTION
**Pull request by ymhari**
## Summary by CodeRabbit

- **New Features**
  - Introduced functionality to handle events related to FDSH PVC requests, including processing and logging results.
  - Added capability to generate transmittable payloads for job processes.
  - Enabled storage and validation of PDM requests and manifest, including the creation of family records and eligible family members.

- **Tests**
  - Added comprehensive RSpec tests for DMF request storage, ensuring successful storage, persistence of family and family members, job existence, and transaction details.
  - Introduced shared context in RSpec for generating complex family hash with eligibility information for testing.

### CODE REVIEW

File: app/operations/fdsh/jobs/generate_individual_dmf_payload.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        return Failure('Transmittable payload cannot be created without a started_at as a Datetime') unless params[:started_at].is_a?(DateTime)
+        return Failure('Transmittable payload cannot be created without a publish_on as a Datetime') unless params[:publish_on].is_a?(DateTime)
+        return Failure('Transmittable payload cannot be created without a job') unless params[:job]
+        return Failure('Transmittable payload cannot be created without a subject') unless params[:subject]
+        return Failure('Transmittable payload cannot be created without a alive status family') unless params[:family_entity]
+        @job = params[:job]
+        @subject = params[:subject]
+
+        Success(params)
+      end
```
[the remark this answers is not in the document]
Comment (ymhari) [reply app/operations/fdsh/jobs/generate_individual_dmf_payload.rb:32]: This is fine.

File: app/operations/fdsh/pvc/dmf/request/store_request.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                                                                           family_entity: @family_entity,
+                                                                           key: :dmf_request,
+                                                                           started_at: DateTime.now,
+                                                                           publish_on: DateTime.now,
+                                                                           correlation_id: @family_entity.hbx_id })
+              errors << result.failure if result.failure?
+            end
+
+            errors.any? ? Failure[errors] : Success("Created transmittable payloads for family #{@family_entity.hbx_id}")
+          end
```
[the remark this answers is not in the document]
Comment (ymhari) [reply app/operations/fdsh/pvc/dmf/request/store_request.rb:104]: We are fine as we using the existing pattern for this.

File: app/operations/fdsh/jobs/generate_individual_dmf_payload.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          status_result = update_status({ job: @job, transmission: @transmission, transaction: @transaction }, :failed,
+                                        "Unable to save transaction with payload")
+        else
+          add_errors({ job: @job, transmission: @transmission, transaction: @transaction },
+                     "Unable to transform payload",
+                     :generate_transmittable_payload)
+          status_result = update_status({ job: @job, transmission: @transmission, transaction: @transaction }, :failed, "Unable to transform payload")
+        end
+        status_result.failure? ? status_result : result
+      end
```
[the remark this answers is not in the document]
Comment (ymhari) [reply app/operations/fdsh/jobs/generate_individual_dmf_payload.rb:76]: This is fine, communicated to dev.

File: spec/operations/fdsh/pvc/dmf/request/store_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  before do
+    @job = Transmittable::Job.where(key: :dmf_request).last
+  end
+
+  it "should be success" do
+    expect(result.success?).to be_truthy
+  end
+
+  it 'should have persitance family and family members' do
```
[the remark this answers is not in the document]
Comment (ymhari) [reply spec/operations/fdsh/pvc/dmf/request/store_request_spec.rb:28]: Will address this in next commit.

File: app/event_source/subscribers/fdsh/dmf_medicaid/dmf_medicare_determination_subscriber.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          else
+            logger.error("Error: :on_requested; failed due to:#{result.inspect}")
+          end
+          ack(delivery_info.delivery_tag)
+        rescue StandardError => e
+          logger.error(
+            "Exception: :on_requested\n Exception: #{e.inspect}" \
+            "\n Backtrace:\n" + e.backtrace.join("\n")
+          )
+          ack(delivery_info.delivery_tag)
```
[the remark this answers is not in the document]
Comment (ymhari) [reply app/event_source/subscribers/fdsh/dmf_medicaid/dmf_medicare_determination_subscriber.rb:27]: Will address this in next commit.

**Review by vkghub — APPROVED** [review]
LGTM

**Review by TristanB17 — APPROVED** [review]
lgtm

**Pull request merged**